### ISSUE TITLE
BUG: Upgrading from Matplotlib 3.5.3 to 3.6.0 Breaks voronoi_plot_2d()

### ISSUE THREAD

**Issue by Chad-Peterson** [scipy/scipy#17173]
### Describe your issue.

Upgrading from Matplotlib 3.5.3 to 3.6.0 Breaks voronoi_plot_2d()

### Reproducing Code Example

```python
rng = np.random.default_rng()
points = rng.random((10,2))

vor = Voronoi(points)

fig = voronoi_plot_2d(vor)

fig = voronoi_plot_2d(vor, show_vertices=False, line_colors='orange',
                line_width=2, line_alpha=0.6, point_size=2)
plt.show()
```

### Error message

```shell
Traceback (most recent call last):
  File "C:\Users\cpgui\AppData\Local\Programs\Python\Python310\lib\code.py", line 90, in runcode
    exec(code, self.locals)
  File "<input>", line 1, in <module>
  File "C:\Program Files\JetBrains\PyCharm 2022.2\plugins\python\helpers\pydev\_pydev_bundle\pydev_umd.py", line 198, in runfile
    pydev_imports.execfile(filename, global_vars, local_vars)  # execute the script
  File "C:\Program Files\JetBrains\PyCharm 2022.2\plugins\python\helpers\pydev\_pydev_imps\_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "C:/Users/cpgui/PycharmProjects/Toyota-Research/main.py", line 34, in <module>
    fig = voronoi_plot_2d(vor)
  File "<decorator-gen-10>", line 2, in voronoi_plot_2d
  File "C:\Users\cpgui\PycharmProjects\Toyota-Research\venv\lib\site-packages\scipy\spatial\_plotutils.py", line 12, in _held_figure
    fig = plt.figure()
  File "C:\Users\cpgui\PycharmProjects\Toyota-Research\venv\lib\site-packages\matplotlib\_api\deprecation.py", line 454, in wrapper
    return func(*args, **kwargs)
  File "C:\Users\cpgui\PycharmProjects\Toyota-Research\venv\lib\site-packages\matplotlib\pyplot.py", line 771, in figure
    manager = new_figure_manager(
  File "C:\Users\cpgui\PycharmProjects\Toyota-Research\venv\lib\site-packages\matplotlib\pyplot.py", line 346, in new_figure_manager
    _warn_if_gui_out_of_main_thread()
  File "C:\Users\cpgui\PycharmProjects\Toyota-Research\venv\lib\site-packages\matplotlib\pyplot.py", line 336, in _warn_if_gui_out_of_main_thread
    if (_get_required_interactive_framework(_get_backend_mod()) and
  File "C:\Users\cpgui\PycharmProjects\Toyota-Research\venv\lib\site-packages\matplotlib\pyplot.py", line 206, in _get_backend_mod
    switch_backend(dict.__getitem__(rcParams, "backend"))
  File "C:\Users\cpgui\PycharmProjects\Toyota-Research\venv\lib\site-packages\matplotlib\pyplot.py", line 266, in switch_backend
    canvas_class = backend_mod.FigureCanvas
AttributeError: module 'backend_interagg' has no attribute 'FigureCanvas'. Did you mean: 'FigureCanvasAgg'?
```

### SciPy/NumPy/Python version information

scipy 1.9.1, numpy 1.23.3, python 3.10

**Comment by tylerjereddy** [comment]
I think this is perhaps a bug in the interaction between Pycharm and `matplotlib` 3.6.0, rather than something we can fix on our end. See: https://github.com/matplotlib/matplotlib/issues/23911

The fix appears to be set for `matplotlib` `3.6.1`.

**Comment by Chad-Peterson** [comment]
Hi Tyler,

Thank you. Yes it looks like matplotlib 3.6.1 was released 16 hours ago and does fix it for Pycharm. Good point with the IDE, I didn't think it would have any bearing on that.

Also confirmed it works on VSCode with the Pycharm-problematic 3.6.0.
